lib/modules/manager/gitlabci/extract.ts

```diff
@@ -122,9 +122,16 @@ function extractDepFromIncludeComponent(
   includeComponent: GitlabIncludeComponent,
   registryAliases?: Record<string, string>,
 ): PackageDependency | null {
-  const componentReference = componentReferenceRegex.exec(
-    includeComponent.component,
-  )?.groups;
+  let componentPath = includeComponent.component;
+  const componentFQDN = componentPath.substring(0, componentPath.indexOf('/'));
```

Comment:
why this substring? makes no sense to me. used should be able to alias more than the first part.

```suggestion
  const componentFQDN = componentPath;
```

Reply:
I don't get it, `componentPath` is always combination of a variable and a path to the specifiek component and version. The reason for the aliases to replace the first path of the string with the absolute URL.
Like in the test:
```
- component: $COMPONENT_REGISTRY/a-project/a-component@1.0
```
You want to get `$COMPONENT_REGISTRY` from the componentPath to check if variable can be found in the `registryAliases` array.

If you want to be more dynamic, then we need to loop through all aliases and check if alias can be found in the component path and replace that. But that was not how the old implementation was working, but can be added.

```
// dummy code
for (registryAliases as key => value} {
     componentPath = componentPath.replace(key, value)
}
```

But maybe I don't understand your suggestion.